tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,760 @@
+	bgpAsn int, nodeAddrList, hubIPAddresses, externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes []string,
+	enableMultiHop, enableBFD bool) *configmap.Builder {
+	frrBFDConfig := frr.DefineBGPConfigWithStaticRouteAndNetwork(
+		bgpAsn, tsparams.LocalBGPASN, hubIPAddresses, externalAdvertisedIPv4Routes,
+		externalAdvertisedIPv6Routes, removePrefixFromIPList(nodeAddrList), enableMultiHop, enableBFD)
+	configMapData := frr.DefineBaseConfig(tsparams.DaemonsFile, frrBFDConfig, "")
+	masterConfigMap, err := configmap.NewBuilder(APIClient, "frr-master-node-config", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+
+	return masterConfigMap
+}
+
+func verifyExternalAdvertisedRoutes(frrPod *pod.Builder, ipv4NodeAddrList, externalExpectedRoutes []string) error {
+	// Get advertised routes from FRR pod, now returned as a map of node IPs to their advertised routes
+	advertisedRoutesMap, err := frr.GetBGPAdvertisedRoutes(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+	if err != nil {
+		return fmt.Errorf("failed to find advertised routes: %w", err)
+	}
+
+	// Iterate through each node in the advertised routes map
+	for nodeIP, actualRoutes := range advertisedRoutesMap {
+		// Split the string of advertised routes into a slice of routes
+		routesSlice := strings.Split(strings.TrimSpace(actualRoutes), "\n")
+
+		// Check that the actual routes for each node contain all the expected routes
+		for _, expectedRoute := range externalExpectedRoutes {
+			matched, err := ContainElement(expectedRoute).Match(routesSlice)
+			if err != nil {
+				return fmt.Errorf("failed to match route %s for node %s: %w", expectedRoute, nodeIP, err)
+			}
+
+			if !matched {
+				return fmt.Errorf("expected route %s not found for node %s", expectedRoute, nodeIP)
+			}
+		}
+	}
+
+	// Return nil if all expected routes are found for all nodes
+	return nil
+}
+
+func resetOperatorAndTestNS() {
+	By("Cleaning MetalLb operator namespace")
+
+	metalLbNs, err := namespace.Pull(APIClient, NetConfig.MlbOperatorNamespace)
+	Expect(err).ToNot(HaveOccurred(), "Failed to pull metalLb operator namespace")
+	err = metalLbNs.CleanObjects(
+		tsparams.DefaultTimeout,
+		metallb.GetBGPPeerGVR(),
+		metallb.GetBFDProfileGVR(),
+		metallb.GetBGPPeerGVR(),
+		metallb.GetBGPAdvertisementGVR(),
+		metallb.GetIPAddressPoolGVR(),
+		metallb.GetMetalLbIoGVR(),
+		metallb.GetFrrConfigurationGVR())
+	Expect(err).ToNot(HaveOccurred(), "Failed to remove object's from operator namespace")
+
+	By("Cleaning test namespace")
+
+	err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+		tsparams.DefaultTimeout,
+		pod.GetGVR(),
+		service.GetServiceGVR(),
+		configmap.GetGVR(),
+		nad.GetGVR())
+	Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+}
+
+func buildRoutesMapWithSpecificRoutes(podList []*pod.Builder, nextHopList []string) (map[string]string, error) {
```

Comment:
Valdate all errors inside the local func via Expect()

Reply:
changed